ethpm/tools/builder.py

```diff
@@ -134,12 +134,22 @@ def get_names_and_paths(compiler_output: Dict[str, Any]) -> Dict[str, str]:
     Return a mapping of contract name to relative path as defined in compiler output.
     """
     return {
-        contract_name: path
+        contract_name: make_path_relative(path)
         for path in compiler_output
         for contract_name in compiler_output[path].keys()
     }
 
 
+def make_path_relative(path: str) -> str:
+    """
+    Returns the given path prefixed with "./" if the path
+    is not already relative in the compiler output.
+    """
+    if path[:2] != "./":
```

Comment:
I can't remember the state of things so this is somewhat homework for you to do.  Can you double-check/verify that we aren't subject to other types of incidental invalid paths being added.  For example, if the path string was `../../Owned.sol` this would produce `./../../Owned.sol` which *should* be invalid since it traverses outside of the *virtual source tree* for this manifest's contracts.

If this is already in place just ignore me my paranoia

Reply:
@pipermerriam Sure, i'll definitely look into it. Can you expand a little on *virtual source tree* - what's inside / outside of the tree?